lib/pure/os.nim

```diff
@@ -1688,13 +1688,28 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo =
     rawToFormalFileInfo(rawInfo, path, result)
 
 proc isHidden*(path: string): bool =
-  ## Determines whether a given path is hidden or not. Returns false if the
-  ## file doesn't exist. The given path must be accessible from the current
-  ## working directory of the program.
-  ##
-  ## On Windows, a file is hidden if the file's 'hidden' attribute is set.
-  ## On Unix-like systems, a file is hidden if it starts with a '.' (period)
-  ## and is not *just* '.' or '..' ' ."
+  ## Determines whether a given path is hidden or not.
+  ## On Windows, a path is hidden if it exists and it's 'hidden' attribute is set.
+  ## On posix, a hidden path starts with `.` (see examples).
+  ## reference: https://en.wikipedia.org/wiki/Hidden_file_and_hidden_directory
+  ## NOTE: on OSX, in addition to the "dotfile" behaviour, existing paths with the
+  ## "Invisible" attribute are hidden in Finder, although not in ls
+  ## NOTE: the behavior of certain corner cases (see examples) ".foo/" is
+  ## unspecified, subject to change (can be avoided using `normalizePath`)
```

Comment:
When writing docs, please check what they look like with ``nim doc``. Currently this is all going to be one huge paragraph, and therefore hard to read. Add empty ``##`` to separate this and use bold instead of ALLCAPS to emphasise.

Reply:
on that note is there a canonical "perfect" function we can use as a template? (that was one of my goals with https://github.com/nim-lang/Nim/pull/8271 btw)

Reply:
Don't have a "perfect" function to link to, but here is how it IMO should be structured:

```nim
proc foo() =
  ## Performs the foo operation.
  ## 
  ## The foo operation is a very important operation
  ## in computer science. It solves all our problems.
  ##
  ## **Warning:** Calling ``foo`` when the ``blarg``
  ## was activated previously may cause the world to
  ## end.
```

Reply:
I've seen both these:
```nim
`foo`
vs
``foo``
```
and both work (and `foo` is simpler and more familiar for ppl familiar w markdown); ok to use single ticks ?

Reply:
Nope, they aren't the same in RST.

Reply:
done

Reply:
> Don't have a "perfect" function to link to, but here is how these should be structured:

add it to contributing.md PR in https://github.com/nim-lang/Nim/pull/8271/commits/b370902f090ec8ec08707d2834ce21a538945439